functorch/csrc/BatchRulesBinaryOps.cpp

```diff
@@ -218,6 +276,10 @@ TORCH_LIBRARY_IMPL(aten, FT_BATCHED_KEY, m) {
   UNARY_POINTWISE(clamp_max);
   POINTWISE_BOXED(clamp_max_);
 
+  VARIADIC_BDIMS_BOXED(_euclidean_dist);
+  BINARY_POINTWISE(_cdist_forward);
```

Comment:
It looks like cdist requires the dtypes of the input tensors to be the same. The batching rule should not be more permissive than the original operator, so we shouldn't use BINARY_POINTWISE here; we probably want a variant of BINARY_POINTWISE that does not do type promotion

![image](https://user-images.githubusercontent.com/5652049/146701152-81099444-f6d5-4554-900b-7f7337b9f514.png)

Reply:
comparison_pointwise_batch_rule seems to not do type promotion, maybe we could reuse that or do some refactoring

Reply:
The error message we see on the image is raised by `matmul` in https://github.com/pytorch/pytorch/blob/8a912014b166c97fcaf2b89e6ff5a466d1a93f7b/aten/src/ATen/native/Distance.cpp#L44 but it should be caught a bit earlier for the second arg (there is a typo in the code, see https://github.com/pytorch/pytorch/pull/70178)